streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -535,7 +535,7 @@ public void onRestoreEnd(final TopicPartition topicPartition, final String store
      */
     public KafkaStreams(final Topology topology,
                         final Properties props) {
-        this(topology.internalTopologyBuilder, new StreamsConfig(props), new DefaultKafkaClientSupplier());
+        this(topology.adjust(new StreamsConfig(props)).internalTopologyBuilder, new StreamsConfig(props), new DefaultKafkaClientSupplier());
```

Comment:
why do we call `adjust` in each constructor? Doing it once in the private constructor should be sufficient?

Reply:
I could be missing something, but the private constructor takes an `InternalTopologyBuilder` instance and `adjust` needs to execute to enable source topic reuse before returning the `InternalTopologyBuilder`.  
But maybe we could shift the `adjust` method to the `InternalTopologyBuilder` thus only need to call once in the private constructor.

Reply:
What I originally thought is exactly as @bbejeck explained. I think changing to `InternalTopologyBuilder` would be a better idea, will try it out.